### ISSUE TITLE
Add test for to_integer(byte b)

### ISSUE THREAD

**Issue by martinmoene** [Microsoft/GSL#330]
In 
```
template <class IntegerType, class = std::enable_if_t<std::is_integral<IntegerType>::value>>
constexpr IntegerType to_integer(byte b) noexcept
{
    return {b};
}
```
the return statement perhaps should be
```
    return static_cast<IntegerType>(b);
```

**Comment by neilmacintosh** [comment]
@martinmoene Sounds reasonable...do you want to offer a PR that makes the change and adds a test?

**Comment by martinmoene** [comment]
Added PR #365.

**Comment by neilmacintosh** [comment]
This has now been fixed by #365. Thanks @martinmoene!
